Source/core/SocketPort.cpp

```diff
@@ -770,7 +775,11 @@ namespace Thunder {
             }
 #endif
 
+#ifdef __APPLE__ 
+            if ((l_Result = ::socket(localNode.Type(), SocketMode(), localNode.Extension())) == INVALID_SOCKET) {
```

Comment:
If on apple SOCK_CLOEXEC does not exist, just define it in portability.h as #define SOCK_CLOEXEC  0 than check all code where for APPLE you where working around it and I guess (at least here) there i no need anymore to work around it.
If possible try to avoid #ifdef <PLATFORM> if possible as it makes code harder to read. If it must, you should but if it is not needed by just defining a simple value not available on the platform, just define it ;-)

Reply:
Updated, but couldn't able to replace #define SOCK_CLOEXEC O_CLOEXEC , since other place we have to use SOCK_CLOEXEC as 0

Reply:
@HaseenaSainul that remark sounds scary, how can it be that in certain situations it takes the value of O_CLOEXEC and in others it has the value 0 ?

Reply:
O_CLOEXEC is used for fcntl() not for socket() case, "int flags = fcntl(l_Result, F_GETFL, 0) | O_CLOEXEC;" So it has to be different, right ?

Reply:
Do not know :-) Everything in linux is a file also sockets. Lets just park this than we will look at it in a second round!

Reply:
Ok, sure